tests/all/unit/api/helpers/test_storage.py

```diff
@@ -40,3 +42,44 @@ def test_create_url(self):
         self.assertEqual(
             expected_file_url, create_url(request_url, '/some/path/image.png')
             )
+
+    def test_generate_hash(self):
+        """Test generation of hash for a key."""
+
+        def patch_settings(settings):
+            settings.return_value = {
+                'secret': 'secret_key'
+            }
+
+        with patch('app.api.helpers.storage.get_settings') as get_settings:
+            patch_settings(get_settings)
+            test_input = 'case1'
+            exepected_output = 'WUFCV0xHVk'
+            actual_output = generate_hash(test_input)
+            self.assertEqual(exepected_output, actual_output)
+            self.assertEqual(len(actual_output), 10)
+
+            test_input = '/static/uploads/pdf/temp/'
+            exepected_output = 'MzRueVhjY0'
+            actual_output = generate_hash(test_input)
+            self.assertEqual(exepected_output, actual_output)
+            self.assertEqual(len(actual_output), 10)
+
+    """Test local file upload."""
+    @patch('app.api.helpers.storage.upload_local')
+    @patch('app.api.helpers.storage.generate_hash', return_value='hash')
+    @patch('app.api.helpers.storage.get_settings', return_value={'static_domain': 'https://next.eventyay.com'})
+    @patch('app.api.helpers.storage.UploadedFile')
+    def test_upload_local(self, uploadedfile_object, settings, generated_hash, uploadlocal):
+        expected_response = 'https://next.eventyay.com/media/upload_key/hash/test.pdf'
+        uploadedfile_object.filename = 'test.pdf'
+
+        app = Flask(__name__)
+        with app.app_context():
```

Comment:
This does not belong in unit test

Reply:
I need to use app context due to which I wrote this test in integration testing. What is the correct way to proceed if I can not write `app.app_context`

Reply:
Move this test to integration

Reply:
Still not moved

Reply:
my bad, I copied the test instead of moving it.